libibverbs/dynamic_driver.c

```diff
@@ -230,6 +231,7 @@ void load_drivers(void)
 			while ((env_name = strsep(&list, ":;")))
 				load_driver(env_name);
 		}
+		free(list);
```

Comment:
There are two calls to strdupa() here and all of them allocate memory, so if you want to be super-pedantic, you will need to call to free after every strdupa.

Reply:
> There are two calls to strdupa() here and all of them allocate memory, so if you want to be super-pedantic, you will need to call to free after every strdupa.

I guess you mean below line226 and line 230. IIUC, these 2 strdupa() are mutually exclusive.

```
222         if (getuid() == geteuid()) {                                            
223                 char *list = NULL, *env_name;                                   
224                                                                                 
225                 if ((env = getenv("RDMAV_DRIVERS"))) {                          
226                         list = strdupa(env);                                    
227                         while ((env_name = strsep(&list, ":;")))                
228                                 load_driver(env_name);                          
229                 } else if ((env = getenv("IBV_DRIVERS"))) {                     
230                         list = strdupa(env);                                    
231                         while ((env_name = strsep(&list, ":;")))                
232                                 load_driver(env_name);                          
233                 }                                                               
234                 free(list);                                                     
235         }   
```

BTW, i think we could simplify the code a abit where there are some duplicated logic.

Reply:
> you will need to call to free after every strdupa.
My bad, i missleading *strdupa* to *strdup*, where strdupa(3) are automatically free

currently, i only keep one patch in this PR.